pkg/loop/internal/chain_reader_test.go

```diff
@@ -0,0 +1,235 @@
+package internal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"sync"
+	"testing"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc/test/bufconn"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/loop/internal/pb"
+
+	"github.com/fxamacker/cbor/v2"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	. "github.com/smartcontractkit/chainlink-common/pkg/types/interfacetests"
+)
+
+func TestVersionedBytesFunctions(t *testing.T) {
+	const unsupportedVer = 25913
+	t.Run("EncodeVersionedBytes unsupported type", func(t *testing.T) {
+		expected := types.InvalidTypeError
+		invalidData := make(chan int)
+
+		_, err := encodeVersionedBytes(invalidData, JSONEncodingVersion2)
+		if err == nil || !errors.Is(err, expected) {
+			t.Errorf("expected error: %s, but got: %v", expected, err)
+		}
+	})
+
+	t.Run("EncodeVersionedBytes unsupported encoding version", func(t *testing.T) {
+		expected := fmt.Errorf("unsupported encoding version %d for data map[key:value]", unsupportedVer)
+		data := map[string]interface{}{
+			"key": "value",
+		}
+
+		_, err := encodeVersionedBytes(data, unsupportedVer)
+		if err == nil || err.Error() != expected.Error() {
+			t.Errorf("expected error: %s, but got: %v", expected, err)
+		}
+	})
+
+	t.Run("DecodeVersionedBytes", func(t *testing.T) {
+		var decodedData map[string]interface{}
+		expected := fmt.Errorf("unsupported encoding version %d for versionedData [97 98 99 100 102]", unsupportedVer)
+		versionedBytes := &pb.VersionedBytes{
+			Version: unsupportedVer, // Unsupported version
+			Data:    []byte("abcdf"),
+		}
+
+		err := decodeVersionedBytes(&decodedData, versionedBytes)
+		if err == nil || err.Error() != expected.Error() {
+			t.Errorf("expected error: %s, but got: %v", expected, err)
+		}
+	})
+}
+
+func TestChainReaderClient(t *testing.T) {
+	RunChainReaderInterfaceTests(t, &interfaceTester{})
+
+	lis := bufconn.Listen(1024 * 1024)
+	s := grpc.NewServer()
+	es := &errorServer{}
+	pb.RegisterChainReaderServer(s, es)
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			panic(err)
+		}
+	}()
+	defer s.Stop()
+	conn := connFromLis(t, lis)
+	client := &chainReaderClient{grpc: pb.NewChainReaderClient(conn)}
+	ctx := context.Background()
+
+	errorTypes := []error{
+		types.InvalidTypeError,
+		types.FieldNotFoundError,
```

Comment:
Under what circumstances does a `FieldNotFoundError` occur?

Reply:
Question for @nolag

Reply:
Sorry, I must not have reverted that change.

Reply:
Removed for this PR, will be re-added later in one of @nolag PR's